core/common/src/main/java/alluxio/heartbeat/SleepingTimer.java

```diff
@@ -84,7 +85,10 @@ public long tick() throws InterruptedException {
 
   @Override
   public void update() {
-    mIntervalSupplier = mIntervalSupplierSupplier.get();
-    mLogger.info("update {} interval supplier.", mThreadName);
+    SleepIntervalSupplier newSupplier = mIntervalSupplierSupplier.get();
+    if (Objects.equals(mIntervalSupplier, newSupplier)) {
+      mIntervalSupplier = newSupplier;
+      mLogger.info("update {} interval supplier.", mThreadName);
```

Comment:
I don't understand - what's the point in updating the supplier if they compare equal?

Reply:
oh, I wrote the condition incorrectly